vendor/bat-native-ledger/src/bat_get_media.cc

```diff
@@ -785,6 +801,36 @@ void BatGetMedia::onGetChannelIdFromUserPage(
   }
 }
 
+void BatGetMedia::onFetchDataFromNonEmbeddable(
+    uint64_t windowId,
+    const ledger::VisitData& visit_data,
+    const std::string& providerType,
+    const uint64_t& duration,
+    const std::string& media_key,
+    const std::string& mediaURL,
+    int response_status_code,
+    const std::string& response,
+    const std::map<std::string, std::string>& headers) {
+  if (response_status_code != 200) {
+    onMediaActivityError(visit_data, providerType, windowId);
+    return;
+  }
+
+  if (providerType == YOUTUBE_MEDIA_TYPE) {
+    std::string publisherName = parsePublisherName(response);
+    std::string publisher_url =
+        getPublisherUrl(parseChannelId(response), providerType);
+
+    ledger_->LoadURL(publisher_url,
```

Comment:
do we need another call? can't we get everything that we need from `visit_data.url` in previous `fetchDataFromUrl`?

Reply:
if we need two calls please change this one to use function `fetchDataFromUrl`

Reply:
We need another call because all data from this point has come from video page. We need to get panel data from publisher channel page here.

Reply:
so which data are you missing on the video page?

Reply:
favicon_url and channel_id

Reply:
Fixed